include/moqt/subscribe_track_handler.h

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+#pragma once
+
+#include <moqt/core/base_track_handler.h>
+#include <moqt/metrics.h>
+#include <moqt/publish_track_handler.h>
+
+namespace moq::transport {
+
+    /**
+     * @brief MOQ track handler for subscribed track
+     *
+     * @details MOQ subscribe track handler defines all track related callbacks and
+     *  functions for subscribe. Track handler operates on a single track (namespace + name).
+     *
+     *  This extends the base track handler to add subscribe handling
+     */
+    class SubscribeTrackHandler : public BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Receive status codes
+         */
+        enum class Error : uint8_t
+        {
+            kOk = 0,
+            kNotAuthorized,
+            kNotSubscribed,
+            kNoData
+        };
+
+        /**
+         * @brief  Status codes for the subscribe track
+         */
+        enum class Status : uint8_t
+        {
+            kOk = 0,
+            kNotConnected,
+            kSubscribeError,
+            kNotAuthorized,
+            kNotSubscribed,
+            kPendingSubscribeResponse,
+            kSendingUnsubscribe // Triggers callbacks to not be called in this state
```

Comment:
There is some ambiguite between error and status. May be i don't fully grasp the intent and I might be wrong

Reply:
The comment actually should have been a proper doxygen comment, which it was missing the `/<`. I have updated both the publish and subscribe track handlers with the below for their relative sending unX states. 

```
///< In this state, callbacks will not be called
```

There is no error anymore, just status now.  The comment indicates that callbacks will not be called during this state. @shenning00  and I discussed this at length on handling tear down asynchronously gracefully for the application. When the client or server app does an `UnsubscribeTrack(...)` the client/server app would not want to receive callbacks anymore for the track that it just Unpublished or Unsubscribed.  This state indicates that unsubscribe (or also in unpublish) is in process and that callbacks will not be called... because that would be just weird to receive callbacks to a track that you are done with.  

We should talk more about this next week.